pkg/providers/tinkerbell/hardware/validator.go

```diff
@@ -93,6 +97,11 @@ func UniqueHostnames() MachineAssertion {
 			return fmt.Errorf("duplicate Hostname: %v", m.Hostname)
 		}
 
+		err := apimachineryvalidation.IsDNS1123Subdomain(m.Hostname)
```

Comment:
This check is already being done on the `Machine`s `Validate` method. I think the additional check you added was when we read in the yamls again - I've not touched that yet. 

Also, side note, this function is self contained and single responsibility. Ideally we shouldn't add to these instead adding new validation functions.

Reply:
I used to think that you have added validation for hardware yaml file and not the generate hardware file. I have corrected the code.